File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)

        ans = 0
        for num in nums_set:
            count = 1
            if num-1 not in nums_set:
                curr = num
                while curr+1 in nums_set:
                    curr += 1
                    count += 1
                ans = max(count, ans)
        
        return ans
        
        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> set(nums.begin(), nums.end());
        int longestStreak = 0;
        for(auto num: set){
            if(set.count(num-1) == 0){
                int curNum = num;
                int streak = 1;
                while(set.count(curNum+1)){
                    curNum++;
                    streak++;
                }
                longestStreak = max(longestStreak, streak);
            }
        }
        return longestStreak;
    }
};